lib/absinthe/subscription/default_document_storage.ex

```diff
@@ -0,0 +1,65 @@
+defmodule Absinthe.Subscription.DefaultDocumentStorage do
+  @behaviour Absinthe.Subscription.DocumentStorage
+
+  @moduledoc """
+  Default document storage for Absinthe. Stores subscription
+  documents and field keys in a Registry process.
+  """
+
+  @impl Absinthe.Subscription.DocumentStorage
+  def child_spec(opts) do
+    Registry.child_spec(opts)
+  end
+
+  @impl Absinthe.Subscription.DocumentStorage
+  def put(storage_process_name, doc_id, doc_value, field_keys) do
+    pdict_add_fields(doc_id, field_keys)
+
+    for field_key <- field_keys do
+      {:ok, _} = Registry.register(storage_process_name, field_key, doc_id)
+    end
+
+    {:ok, _} = Registry.register(storage_process_name, doc_id, doc_value)
+  end
+
+  @impl Absinthe.Subscription.DocumentStorage
+  def delete(storage_process_name, doc_id) do
+    for field_key <- pdict_fields(doc_id) do
+      Registry.unregister(storage_process_name, field_key)
+    end
+
+    pdict_delete_fields(doc_id)
+
+    Registry.unregister(storage_process_name, doc_id)
+
+    :ok
+  end
+
+  @impl Absinthe.Subscription.DocumentStorage
+  def get_docs_by_field_key(storage_process_name, field_key) do
+    storage_process_name
+    |> Registry.lookup(field_key)
+    |> MapSet.new(fn {_pid, doc_id} -> doc_id end)
+    |> Enum.reduce(%{}, fn doc_id, acc ->
+      case Registry.lookup(storage_process_name, doc_id) do
+        [] ->
+          acc
+
+        [{_pid, doc} | _rest] ->
+          Map.put_new(acc, doc_id, doc)
+      end
+    end)
+  end
```

Comment:
Question, nonblocking:
I know you were porting over the code that had already existed into this behavior, but it strikes me as odd that there is no call to access the fields (`pdict_fields/1`) within the process dictionary in this function. I'm not sure why we add fields to the process dictionary if we don't ever read them directly. Do the `Registry` functions access them in a different way? I thought Registry would be reading info off of ETS.

Reply:
It is used in the `delete` function. I think the idea was to hold a reference to them so they could be removed later when an unsubscribe happen.

Reply:
I saw that, but if they're written and deleted but never read, what are we doing?